chips/stm32f4xx/src/clocks/pll.rs

```diff
@@ -202,6 +206,29 @@ impl<'a, PllConstants: clock_constants::PllConstants> Pll<'a, PllConstants> {
         unreachable!("The previous for loop should always return");
     }
 
+    /// Get the PLL source clock frequency in MHz
+    ///
+    /// # Returns
+    ///
+    /// + [HSI_FREQUENCY_MHZ]: if the pll source is HSI
+    /// + HSE frequency if the pll source is HSE
+    fn get_pll_source_frequency(&self) -> usize {
+        if self.rcc.is_hsi_clock_system_clock() {
+            HSI_FREQUENCY_MHZ
+        } else {
+            self.rcc.get_hse_frequency().unwrap()
+        }
+    }
+
+    /// Set the PLL source clock according to the system clock
+    fn set_pll_source_clock(&self) {
+        if self.rcc.is_hsi_clock_system_clock() {
+            self.rcc.set_pll_clocks_source(PllSource::HSI);
+        } else {
+            self.rcc.set_pll_clocks_source(PllSource::HSE);
+        }
+    }
```

Comment:
I would prefer this function to be `set_pll_source_clock(&self, source: PllSource) -> Result<(), ErrorCode>`. This would set the PLL source clock to the source specified by the parameter `source`.

To change the PLL clock source, the PLL clock has to be disabled, so it can return `ErrorCode::FAIL` if the PLL clock is enabled. Ensuring PLL clock is disabled before changing its source avoids glitches.

What do you think?